tmva/sofie_parsers/src/RModelParser_ONNX.cxx

```diff
@@ -6,1658 +6,212 @@
+extern ParserFuncSignature ParseReduceSumsquare;
+extern ParserFuncSignature ParseReduceProd;
+// Others
+extern ParserFuncSignature ParseBatchNormalization;
+extern ParserFuncSignature ParseTranspose;
+extern ParserFuncSignature ParseRelu;
+extern ParserFuncSignature ParseTanh;
+extern ParserFuncSignature ParseConv;
+extern ParserFuncSignature ParseConvTranspose;
+extern ParserFuncSignature ParseLeakyRelu;
+extern ParserFuncSignature ParseSelu;
+extern ParserFuncSignature ParseSigmoid;
+extern ParserFuncSignature ParseGemm;
+extern ParserFuncSignature ParseRNN;
+extern ParserFuncSignature ParseLSTM;
+extern ParserFuncSignature ParsePool;
+extern ParserFuncSignature ParseReshape;
+extern ParserFuncSignature ParseSlice;
+extern ParserFuncSignature ParseGRU;
+extern ParserFuncSignature ParseIdentity;
+extern ParserFuncSignature ParseSoftmax;
+extern ParserFuncSignature ParseMax;
+extern ParserFuncSignature ParseConcat;
+extern ParserFuncSignature ParseCast;
+extern ParserFuncSignature ParseShape;
+// Decalaration of fused operators
+extern ParserFuseFuncSignature ParseFuseConvAdd;
+extern ParserFuseFuncSignature ParseFuseConvTransposeAdd;
+extern ParserFuseFuncSignature ParseFuseMatMulAdd;
+
+// Parse an operator
+std::unique_ptr<ROperator>
+RModelParser_ONNX::ParseOperator(const size_t i, const onnx::GraphProto &graphproto, const std::vector<size_t> &nodes)
 {
+   int idx = (nodes.size() > i) ? nodes[i] : (int)i;
+   const auto &nodeproto = graphproto.node(idx);
+   const std::string op_type = nodeproto.op_type();
+   if (fVerbose)
+      std::cout << "Parsing an operator " << op_type << std::endl;
 
-   ETensorType input_type;
-
-   auto input_name = nodeproto.input(0);
-   auto it = tensor_type.find(input_name);
-   if (it != tensor_type.end()) {
-      input_type = it->second;
-   } else {
-      throw std::runtime_error("TMVA::SOFIE ONNX Parser BatchNorm op has input tensor " + input_name +
-                               " but its type is not yet registered");
-      // std::cout << "TMVA::SOFIE ONNX Parser " << nodeproto.op_type() <<
-      // " op has input tensor" + input_name + "  but its type is not yet registered - use default float " << std::endl;
-      // input_type = ETensorType::FLOAT;
+   auto it = fOperatorsMapImpl->fOperatorsMap.find(op_type);
+   if (it == fOperatorsMapImpl->fOperatorsMap.end()) {
+      throw std::runtime_error("TMVA::SOFIE Operator type " + op_type + " is not yet supported");
    }
 
-   std::unique_ptr<ROperator> op;
-   float fepsilon = 1e-05;
-	float fmomentum = 0.9;
-	std::size_t ftraining_mode = 0;
-
-   switch(input_type) {
-      case ETensorType::FLOAT:
-         if (nodeproto.input_size() == 5) {
-            op.reset(new ROperator_BatchNormalization<float>(fepsilon, fmomentum, ftraining_mode, nodeproto.input(0), nodeproto.input(1), nodeproto.input(2), nodeproto.input(3), nodeproto.input(4), nodeproto.output(0)));
+   // auto find = mapOptypeOperator.find(nodeproto.op_type());
```

Comment:
Can we remove the commented code if they aren't required anymore?

Reply:
Done. Thank you @sanjibansg for the review!